gef-ui/server/controllers/companies-house/index.js

```diff
@@ -24,46 +23,22 @@ const companiesHouse = async (req, res) => {
 };
 
 const validateCompaniesHouse = async (req, res) => {
-  const { params, body, query, session } = req;
-  const { regNumber } = body;
-  const { dealId } = params;
-  const { status } = query;
-  const { user, userToken } = session;
-  const { _id: editorId } = user;
-  const companiesHouseErrors = [];
-  let companiesHouseDetails;
-
-  if (!regNumber) {
-    companiesHouseErrors.push({
-      errRef: 'regNumber',
-      errMsg: 'Enter a Companies House registration number',
-    });
-  }
-
-  if (regNumber && !isValidCompaniesHouseNumber(regNumber)) {
-    companiesHouseErrors.push({
-      errRef: 'regNumber',
-      errMsg: 'Enter a valid Companies House registration number',
-    });
-  }
-
   try {
-    const { exporter } = await api.getApplication({ dealId, userToken });
+    const { params, body, query, session } = req;
+    const { regNumber: registrationNumber } = body;
+    const { dealId } = params;
+    const { status } = query;
+    const { user, userToken } = session;
+    const { _id: editorId } = user;
 
-    if (companiesHouseErrors.length === 0) {
```

Comment:
Can we add `registrationNumber` void check before any further API calls please.

Reply:
I've moved the call to `api.getApplication` to after the call to `api.getCompanyByRegistrationNumber`. `api.getCompanyByRegistrationNumber` does a validation check at the very start before making the API call, so now if `registrationNumber` is invalid, we avoid making any API calls.

Reply:
Added assertion to ensure no further API calls are made

Reply:
Thanks